src/main/java/org/assertj/core/api/Assumptions.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2017 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import net.sf.cglib.proxy.Enhancer;
+import net.sf.cglib.proxy.MethodInterceptor;
+import net.sf.cglib.proxy.MethodProxy;
+import org.assertj.core.util.CheckReturnValue;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+public class Assumptions {
+
+  /**
+   * Creates a new instance of <code>{@link ObjectAssert}</code> assumption.
+   *
+   * @param actual the actual value.
+   * @param <T> the type of the actual value.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  @SuppressWarnings("unchecked")
+  public static <T> AbstractObjectAssert<?, T> assumeThat(T actual) {
+    return (AbstractObjectAssert<?, T>) enhanceAsAssumption(ObjectAssert.class, Object.class, actual);
+  }
+
+  /**
+   * Creates a new instance of <code>{@link StringAssert}</code> assumption.
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, String> assumeThat(String actual) {
+    return enhanceAsAssumption(StringAssert.class, String.class, actual);
+  }
+
+  @SuppressWarnings("unchecked")
+  private static <ASSERTION, ACTUAL> ASSERTION enhanceAsAssumption(Class<ASSERTION> assertionType, Class<ACTUAL> actualType,
+                                                                   Object actual) {
+    Enhancer enhancer = new Enhancer();
+    enhancer.setSuperclass(assertionType);
+    enhancer.setCallback(new MethodInterceptor() {
+      @Override
+      public Object intercept(Object o, Method method, Object[] args, MethodProxy methodProxy) throws Throwable {
+        try {
+          Object result = methodProxy.invokeSuper(o, args);
+          if (result != o && result instanceof AbstractAssert) {
+            return enhanceAsAssumption(result);
+          }
+          return result;
+        } catch (AssertionError e) {
+          throw (RuntimeException) Class.forName("org.junit.AssumptionViolatedException")
+                                        .getConstructor(String.class, Throwable.class).newInstance("", e);
```

Comment:
Here I use reflection to create JUnit's `AssumptionViolatedException`. Probably it is an overdo, as JUnit dependency scope is `provided` at pom.xml. But as it was requested to avoid hard dependency on JUnit, implemented it this way.